### ISSUE TITLE
Incompatible with Firefox e10s (Electrolysis)

### ISSUE THREAD

**Issue by tonycpsu** [Akimkin/autogroup#10]
I'm interested in using Firefox [e10s](https://wiki.mozilla.org/Electrolysis) for better performance and isolation of problematic tabs, but the AutoGroup add-on doesn't seem to work when e10s is enabled (by setting `browser.tabs.remote.autostart;true` in `about:config` using a recent beta or nightly.  (I'm using 44.0b9 on OSX.)  I've tried 0.3.1 and 0.4 and neither work -- tabs simply load in whatever tab group they're in.

**Comment by Akimkin** [comment]
Yep,  some time ago I got a letter from Mozilla informing about changes in Electrolysis that would probably make this addon incompatible.
It should be possible to fix this (Mozilla made a guide on porting).
Gonna look into that.

**Comment by tonycpsu** [comment]
Great, glad to see it's on your radar..  For now the performance improvement of e10s seems signfiicant enough that I'm going to have to go without autogroup until this is resolved.

**Comment by Akimkin** [comment]
Ok, I think I've found the solution.
Please, try testing dev-e10s branch. Although it's based on v0.4, which has problems with MacOS, it shouldn't be too hard to backport [this patch] (https://github.com/Akimkin/autogroup/commit/742936a9e97af14273b62f95c763e774eee60447) to v0.3.1.

P.S: [Some bad news from Mozilla](https://support.mozilla.org/en-US/kb/tab-groups-removal).
It seems that Panorama's functionality will be provided by the [Tab-Groups addon](https://addons.mozilla.org/en-US/firefox/addon/tab-groups-panorama/), so AutoGroup will become kind of "addon for an addon" :\

**Comment by tonycpsu** [comment]
Yes, I'm aware of the removal of Panorama, so I installed Tab Groups a while ago.  I understand why they took it out, but I still find it very useful and hope it doesn't become deprecated.

I tried your e10s branch and it does seem to work, though as you say, since I'm on a Mac, I can't edit my tab groups unless I downgrade to 0.3.  I for one have no problems using your dev branch for now and waiting for a fix for #9 whenever you get around to it.

**Comment by Akimkin** [comment]
Both master and Electrolysis branches are updated with fix for erroneous behavior of config dialog on OSX. Please, test.

**Comment by tonycpsu** [comment]
Looks great to me in e10s.  Thanks much!

**Comment by tonycpsu** [comment]
Upon further review...  I can see all my groups and rules now, but when I try to add a new rule, it doesn't allow me to save.  I don't know if I just forgot to test this or if a recent Firefox change broke it. I'm on 45.0 right now,

**Comment by Akimkin** [comment]
That's probably an UI thing. Sometimes settings dialog has not enough height, so its lower part is just not visible. I pushed [this commit](https://github.com/Akimkin/autogroup/commit/0ed723cb633608a8c2f3582a3063f1a2c640adf5) in attempt to fix this.

I think that current UI should be refactored completely. A see As an example of new UI 
That's probably an UI thing. Sometimes settings dialog has not enough height, so its lower part is just not visible. I pushed [this commit](https://github.com/Akimkin/autogroup/commit/0ed723cb633608a8c2f3582a3063f1a2c640adf5) in attempt to fix this. Please, try it out.

I think that it will be better to update current Preferences UI to something like the Tab Groups addon has (i.e., replace the separate Preferences dialog with a preferences UI in a browser tab). I may look into that later.

**Comment by adipose** [comment]
Any chance of release a signed XPI with this dev fix? e10s is now in release mode and I cannot use it with this addon :)

**Comment by Akimkin** [comment]
Not a big problem to release a fix (I still didn't updated to FF54). I'll probably do it in some upcoming days,

But I really doubt further development.
At first, Mozilla removed Panorama, but Tab Groups addon appeared to help. But now Mozilla changes the whole system of extensions and author of Tab Groups made a statement that his addon will stop working since FF57 and will be discontinued since it would be unlikely to reimplement it on top of new extension system. Same will probably happen to Auto Group if no suitable alternative for tab management will appear :(

**Comment by adipose** [comment]
Yeah, my hope is that there will be some delays or at least continued compatibility of old addons due to the infeasibility of dropping so many.  But in any case, I will use tab groups as long as I can--but I'd like to use e10s as soon as I can :)
